d3a_interface/utils.py

```diff
@@ -58,6 +57,45 @@ def generate_market_slot_list_from_config(sim_duration, start_date, market_count
         if (slot_length * i) <= sim_duration]
 
 
+def generate_market_slot_list():
+    start_date = today(tz=TIME_ZONE) \
+        if GlobalConfig.IS_CANARY_NETWORK else GlobalConfig.start_date
+    time_span = duration(days=CN_PROFILE_EXPANSION_DAYS) \
+        if GlobalConfig.IS_CANARY_NETWORK else GlobalConfig.sim_duration
+    sim_duration_plus_future_markets = time_span + GlobalConfig.slot_length * \
+        (GlobalConfig.market_count - 1)
+    market_slot_list = \
+        generate_market_slot_list_from_config(sim_duration=sim_duration_plus_future_markets,
+                                              start_date=start_date,
+                                              market_count=GlobalConfig.market_count,
+                                              slot_length=GlobalConfig.slot_length)
+
+    if not getattr(GlobalConfig, 'market_slot_list', []):
+        GlobalConfig.market_slot_list = market_slot_list
+    return market_slot_list
+
+
+def find_object_of_same_weekday_and_time(indict, time_slot, ignore_not_found=False):
```

Comment:
Thanks a lot for the refactoring, as we are touching these functions it would be great if we can add small documentation of what it does or at least the params/ return types

Reply:
Just did it for the longest functions in the module, feel free to let me know if something is not clear, thanks!

Reply:
all clear, much appreciated !